PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,557 @@
+            self.insertElem(val)
+
+    def clickDelete(self):
+        val = self.validateArgument()
+        if not self.first:
+            msg = "ERROR: Linked list is empty"
+        elif not val:
+            return
+        else:
+            result = self.delete(val)
+            if result != None:
+                msg = "{} Deleted!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+        self.setMessage(msg)
+        self.clearArgument()
+        
+    def clickDeleteFirst(self):
+        if not self.first: 
+            msg = "ERROR: Linked list is empty"
+        else:
+            self.deleteFirst()
+            msg = "first node deleted"
+        self.setMessage(msg)
+        self.clearArgument()
+        
+        
+    def clickNewLinkedList(self):
+        self.canvas.delete('all')
+        self.newLinkedList()
+    
+    def clickGetFirst(self):
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
+    
+
+    def makeButtons(self):
+        vcmd = (self.window.register(validate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        searchButton = self.addOperation(
+            "Search", lambda: self.clickSearch(), numArguments=1,
+            validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1,
+            validationCmd=vcmd)
+        deleteButton = self.addOperation(
+            "Delete", lambda: self.clickDelete(), numArguments=1,
+            validationCmd=vcmd)
+        self.addAnimationButtons()
+        deleteFirstButton = self.addOperation("Delete First", lambda: self.clickDeleteFirst(), 
+                                              numArguments = 0, validationCmd = vcmd, maxRows = 3)
+        newLinkedListButton = self.addOperation(
+            "New", lambda: self.clickNewLinkedList(), 
+            numArguments = 0, validationCmd =vcmd, maxRows = 3)
+        getFirstButton = self.addOperation(
+            "Get First", lambda: self.clickGetFirst(), numArguments = 0,
+            validationCmd=vcmd, maxRows = 3)
+
+    
+        return [searchButton, insertButton, deleteButton, deleteFirstButton, newLinkedListButton, getFirstButton]        
+
+    
+    # validate text entry
+    def validateArgument(self):
+        entered_text = self.getArgument()
+        if entered_text:
+            if len(entered_text) <= self.maxArgWidth:
+                return entered_text
+            else:
+                self.setMessage("Error! {} value is too long".format(entered_text))    
+
+            
+##allow letters or numbers to be typed in                  
+def validate(action, index, value_if_allowed,
+            prior_value, text, validation_type, trigger_type, widget_name):
+    return True
```

Comment:
If validate is always going to return `True`, then there's no point in setting a validation function for the text entry widget.  My [earlier comment](https://github.com/JMCanning78/datastructures-visualization/pull/112#discussion_r468246363) suggested preventing the length of the entered string from going beyond the maximum argument width.  That could be done by returning `len(value_if_allowed) <= self.maxArgWidth` on line 543.